lib/ansible/modules/cloud/amazon/route53_facts.py

```diff
@@ -338,9 +575,10 @@ def record_sets_details(client, module):
         module.fail_json(msg="start_record_name must be specified if type is set")
     elif module.params.get('type'):
         params['StartRecordType'] = module.params.get('type')
-
-    results = client.list_resource_record_sets(**params)
-    return results
+    try:
+        return list_record_sets_with_backoff(client, params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't list record sets")
```

Comment:
If providing a hosted_zone_id that has been removed, this fails. Should it fail, or should record sets just be an empty dict if e.response['Error']['Code'] == 'NoSuchHostedZone' in list_record_sets_with_backoff()? Wondered something similar throughout reading this PR in a number of other places too.

Reply:
Thanks for this spot @s-hertel - it's a great catch. To be honest I think it's a flaw with AWSRetry that this happens - retries for not found should be handled by calling functions. 

But until there's consensus on that, I'll add a fix for this.